src/Symfony/Component/Form/FormConfigBuilder.php

```diff
@@ -915,6 +915,6 @@ public static function validateName($name)
      */
     public static function isValidName($name)
     {
-        return '' === $name || null === $name || preg_match('/^[a-zA-Z0-9_][a-zA-Z0-9_\-:]*$/D', $name);
+        return '' === $name || null === $name || preg_match('/^\w[\w\-:]*$/D', $name);
```

Comment:
\w and \W are locale aware (also Unicode-aware if we'd add u-modifier), but taking into account that \w is used in Symfony code base it should not be a problem (or this got escalated already, considering the amount of project built on it).

Regarding readability, IMO using existing character classes is more expressive regarding regex intention. There are of course cases (like CC validation) must be done with implicit characters set declaration, but changed cases looking more expressive now (otherwise why I should invest time into modifying them ;) ).